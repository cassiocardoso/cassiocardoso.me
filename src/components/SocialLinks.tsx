import * as React from 'react';
import styled from 'styled-components';
import Github from 'react-feather/dist/icons/github';
import Linkedin from 'react-feather/dist/icons/linkedin';
import Twitter from 'react-feather/dist/icons/twitter';
import Codepen from 'react-feather/dist/icons/codepen';

const StyledSocialLinks = styled.ul`
  display: flex;
  list-style: none;
  margin: 0;
  padding: 0;

  li {
    margin: 0;
  }

  a {
    margin: 1rem;
    padding: 1rem;

    &:first-child {
      margin-left: 0;
      padding-left: 0;
    }
  }
`;

const SocialLinks = () => (
  <StyledSocialLinks>
    <li>
      <a
        href="https://linkedin.com/in/cassiocardoso"
        target="_blank"
        rel="noopener noreferrer"
      >
        <Linkedin />
      </a>
    </li>
    <li>
      <a
        href="https://github.com/cassiocardoso"
        target="_blank"
        rel="noopener noreferrer"
      >
        <Github />
      </a>
    </li>
    <li>
      <a
        href="https://codepen.io/cassiocardoso"
        target="_blank"
        rel="noopener noreferrer"
      >
        <Codepen />
      </a>
    </li>
    <li>
      <a
        href="https://twitter.com/cassiocardoso"
        target="_blank"
        rel="noopener noreferrer"
      >
        <Twitter />
      </a>
    </li>
  </StyledSocialLinks>
);

export default SocialLinks;
