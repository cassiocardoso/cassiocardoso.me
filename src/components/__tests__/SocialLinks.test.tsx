import * as React from 'react';
import { render } from 'react-testing-library';

import SocialLinks from '../SocialLinks';

describe('<SocialLinks />', () => {
  it('should render correctly', () => {
    const { container } = render(<SocialLinks />);

    expect(container).toMatchSnapshot();
  });
});
