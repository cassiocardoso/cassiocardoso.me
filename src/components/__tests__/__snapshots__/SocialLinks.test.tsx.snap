// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SocialLinks /> should render correctly 1`] = `
<div>
  <ul
    class="sc-bdVaJa huQEoI"
  >
    <li>
      <a
        href="https://linkedin.com/in/cassiocardoso"
        rel="noopener noreferrer"
        target="_blank"
      >
        <svg
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16 8a6 6 0 0 1 6 6v7h-4v-7a2 2 0 0 0-2-2 2 2 0 0 0-2 2v7h-4v-7a6 6 0 0 1 6-6z"
          />
          <rect
            height="12"
            width="4"
            x="2"
            y="9"
          />
          <circle
            cx="4"
            cy="4"
            r="2"
          />
        </svg>
      </a>
    </li>
    <li>
      <a
        href="https://github.com/cassiocardoso"
        rel="noopener noreferrer"
        target="_blank"
      >
        <svg
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9 19c-5 1.5-5-2.5-7-3m14 6v-3.87a3.37 3.37 0 0 0-.94-2.61c3.14-.35 6.44-1.54 6.44-7A5.44 5.44 0 0 0 20 4.77 5.07 5.07 0 0 0 19.91 1S18.73.65 16 2.48a13.38 13.38 0 0 0-7 0C6.27.65 5.09 1 5.09 1A5.07 5.07 0 0 0 5 4.77a5.44 5.44 0 0 0-1.5 3.78c0 5.42 3.3 6.61 6.44 7A3.37 3.37 0 0 0 9 18.13V22"
          />
        </svg>
      </a>
    </li>
    <li>
      <a
        href="https://codepen.io/cassiocardoso"
        rel="noopener noreferrer"
        target="_blank"
      >
        <svg
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polygon
            points="12 2 22 8.5 22 15.5 12 22 2 15.5 2 8.5 12 2"
          />
          <line
            x1="12"
            x2="12"
            y1="22"
            y2="15.5"
          />
          <polyline
            points="22 8.5 12 15.5 2 8.5"
          />
          <polyline
            points="2 15.5 12 8.5 22 15.5"
          />
          <line
            x1="12"
            x2="12"
            y1="2"
            y2="8.5"
          />
        </svg>
      </a>
    </li>
    <li>
      <a
        href="https://twitter.com/cassiocardoso"
        rel="noopener noreferrer"
        target="_blank"
      >
        <svg
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M23 3a10.9 10.9 0 0 1-3.14 1.53 4.48 4.48 0 0 0-7.86 3v1A10.66 10.66 0 0 1 3 4s-4 9 5 13a11.64 11.64 0 0 1-7 2c9 5 20 0 20-11.5a4.5 4.5 0 0 0-.08-.83A7.72 7.72 0 0 0 23 3z"
          />
        </svg>
      </a>
    </li>
    <li>
      <a
        href="https://instagram.com/cassiocardoso"
        rel="noopener noreferrer"
        target="_blank"
      >
        <svg
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            height="20"
            rx="5"
            ry="5"
            width="20"
            x="2"
            y="2"
          />
          <path
            d="M16 11.37A4 4 0 1 1 12.63 8 4 4 0 0 1 16 11.37z"
          />
          <line
            x1="17.5"
            x2="17.5"
            y1="6.5"
            y2="6.5"
          />
        </svg>
      </a>
    </li>
  </ul>
</div>
`;
